src/GitTfs.VsCommon/TfsHelper.Vs2017Base.cs

```diff
@@ -61,7 +72,51 @@ protected override bool HasWorkItems(Changeset changeset)
         /// </returns>
         protected string GetVsInstallDir()
         {
-            return @"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise";
+            if (myVisualStudioInstallationPath != null)
+                return myVisualStudioInstallationPath;
+
+            var setupConfiguration = (ISetupConfiguration2)GetSetupConfiguration();
+            IEnumSetupInstances instancesEnumerator = setupConfiguration.EnumAllInstances();
+
+            int fetched;
+            var instances = new ISetupInstance[1];
+            while (true)
+            {
+                instancesEnumerator.Next(1, instances, out fetched);
+                if (fetched <= 0)
+                    break;
+
+                var instance = (ISetupInstance2)instances[0];
```

Comment:
Are there cases where this cast would fail?

Reply:
Not that I am aware of. But if it does, AFAIR we fail to instantiate the TfsHelper which would result in skipping the concrete plugin, e.g. we would think that VS2017 support is working.